src/omotes_simulator_core/entities/network_controller.py

```diff
@@ -52,8 +69,9 @@ def update_setpoints(self, time: datetime.datetime) -> dict:
         :return: dict with the key the asset id and the heat demand for that asset.
         """
         # TODO add also the possibility to return mass flow rate instead of heat demand.
-        if (self.get_total_supply() + (
-                -1 * self.get_total_discharge_storage())) <= self.get_total_demand(time):
+        if (
```

Comment:
Suggestion for a next PR: split this method in a _update_setpoints_constrained_demand and _update_setpoints_available_demand